push/sender/src/main/java/org/jboss/aerogear/unifiedpush/message/sender/APNsPushNotificationSender.java

```diff
@@ -209,4 +205,43 @@ public void messageSendFailed(ApnsNotification message, Throwable e) {
         // null if, why ever, there was no cert/passphrase
         return null;
     }
+
+    /**
+     * Configure the Gateway to the Apns servers.
+     * Default gateway and port can be override with respectively :
+     *  - aerogear.apns.push.host
+     *  - aerogear.apns.push.port
+     *
+     * Feedback gateway and port can be override with  respectively :
+     *  - aerogear.apns.feedback.host
+     *  - aerogear.apns.feedback.port
+     * @param iOSVariant
+     * @param builder
+     */
+    private void configureDestinations(iOSVariant iOSVariant, ApnsServiceBuilder builder) {
+        // pick the destination:
+        if (iOSVariant.isProduction()) {
+            builder.withProductionDestination();
+        } else {
+            builder.withSandboxDestination();
+        }
+
+        //Is the gateway host&port provided by a system property ?
+        if(System.getProperty("aerogear.apns.push.host")!= null){
+            int port = Utilities.SANDBOX_GATEWAY_PORT;
+            if(System.getProperty("aerogear.apns.push.port")!= null) {
```

Comment:
+1 on using a constant - thought the same when looking at the PR :-)

On Wed, Jan 21, 2015 at 4:36 PM, Tadeas Kriz <notifications@github.com>
wrote:

> In
> push/sender/src/main/java/org/jboss/aerogear/unifiedpush/message/sender/APNsPushNotificationSender.java
> <https://github.com/aerogear/aerogear-unifiedpush-server/pull/470#discussion_r23307178>
> :
>
> > +     *  - aerogear.apns.feedback.port
> > +     * @param iOSVariant
> > +     * @param builder
> > +     */
> > +    private void configureDestinations(iOSVariant iOSVariant, ApnsServiceBuilder builder) {
> > +        // pick the destination:
> > +        if (iOSVariant.isProduction()) {
> > +            builder.withProductionDestination();
> > +        } else {
> > +            builder.withSandboxDestination();
> > +        }
> > +
> > +        //Is the gateway host&port provided by a system property ?
> > +        if(System.getProperty("aerogear.apns.push.host")!= null){
> > +            int port = Utilities.SANDBOX_GATEWAY_PORT;
> > +            if(System.getProperty("aerogear.apns.push.port")!= null) {
>
> Might be a good thing to take these strings and put them as constants at
> the top of this file. Firstly it would be easier to change in the future
> (if needed) and secondly it will be easily seen on top of the file. But
> I'll leave that decision to you :).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/aerogear/aerogear-unifiedpush-server/pull/470/files#r23307178>
> .
>



-- 
Matthias Wessendorf

blog: http://matthiaswessendorf.wordpress.com/
sessions: http://www.slideshare.net/mwessendorf
twitter: http://twitter.com/mwessendorf